.github/workflows/ci.yml

```diff
@@ -74,6 +81,16 @@ env:
 # It adds overhead to the build and another point of failure.
 
 jobs:
+  # List of basic checks that needs to pass before we fire heavier jobs.
+  basics:
+    runs-on: ubuntu-latest
+    needs:
+      - fmt
+      - doctest
+      - check
+    steps:
+      - run: exit 0
```

Comment:
The only jobs that we need to be blocked on this is the `test` build, the rest can fire on failure. All the regular runners we can just fire off aggressively, trying to reduce CI latency.